ontologyvalidator/ontologyvalidateutil.py

```diff
@@ -0,0 +1,77 @@
+    Given a metadata_document dictionary(i.e JSON document), returns a list of pairs/2-tuples: the first
+    element of the pair is a (pseudo)JSONPaths to an ontology term in the document and the second element of
+    is the ontology term itself.
+    e.g
+    given a document...
+
+    {
+        "xxx": {
+            "yyy": {
+                "ontology" : "ABC0001234"
+            },
+            "zzz" : {
+                "aaa" : {
+                    "ontology" : "ABC999444"
+                }
+            }
+        }
+    }
+
+    ...returns the list of pairs [("xxx.yyy.ontology", "ABC0001234"),("xxx.zzz.aaa.ontology", "ABC999444")]
+
+    '''
+    def find_ontology_terms_in_document(self, metadata_document: dict):
+        return [item for item in flatten_json.flatten(metadata_document, ".").items() if item[0].endswith(".ontology")]
+
+    '''
+    given a string of form xxx.yyy.zzzzz.ontology, returns zzzzz_ontology.json 
+    '''
+    def get_ontology_schema_file_name_from_ontology_field(self, ontology_field_path: str):
+        split_ontology_field_path = ontology_field_path.split(".")
+        if split_ontology_field_path[-1] != "ontology":
+            raise("Error: " + ontology_field_path + "does not specify a valid ontology field")
+        else:
+            return split_ontology_field_path[-2] + "_" + "ontology" + ".json"
+
+    '''
+    fetches an ontology schema at the specified location
+    '''
+    def retrieve_ontology_schema(self, base_url, filename):
+        return requests.get(base_url + "/" + filename).json()
+
+    '''
+    generates an OLS query dict() given the ontology schema describing graph restrictions and ontology term
+    '''
+    def generate_ols_query(self, ontology_schema: dict, ontology_term: str):
+        try:
+            graph_restrictions = ontology_schema["properties"]["ontology"]["graph_restriction"]
+            # given the list ["obo:uberon", "obo:efo"], returns the list ["uberon", "ufo"]
+            ontologies_to_query = [ontology.split(":")[1] for ontology in graph_restrictions["ontologies"]]
+            # given the list ["uberon", "ufo", "ontologyX", "ontologyY"], returns the string "uberon,ufo,ontologyX,ontologyY"
+            ontologies_to_query_string = reduce(lambda ontology, another_ontology: ontology + "," + another_ontology, ontologies_to_query)
+            ontology_classes_to_query = [ontology_class.replace(":", "_") for ontology_class in graph_restrictions["classes"]]
+            ontology_classes_uris = [self.get_iri_for_ontology_class(ontology_class) for ontology_class in ontology_classes_to_query]
+
+            query_dict = dict()
+            query_dict["q"] = ontology_term
+            query_dict["queryFields"] = "short_form"
+            query_dict["ontology"] = ontologies_to_query_string
+            query_dict["childrenOf"] = reduce(lambda ontology_class_iri, another_ontology_class_iri: ontology_class_iri + "," + another_ontology_class_iri, ontology_classes_uris)
```

Comment:
Can you use `allChildrenOf` by default instead of `childrenOf`. This is more inclusive and will be needed for anatomy validation where some of the children are related via a part_of relation instead of a parent/child